### PR TITLE
feat: generalize coordinate inputs

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "satellite-visualizer",
-  "version": "1.4.3",
+  "version": "1.5.0",
```
Comment (vishwa2710) [inline package-lock.json:3]:
```suggestion
  "version": "1.4.3",
```

File: package-lock.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
   "lockfileVersion": 3,
   "requires": true,
   "packages": {
     "": {
       "name": "satellite-visualizer",
-      "version": "1.4.3",
+      "version": "1.5.0",
```
Comment (vishwa2710) [inline package-lock.json:9]:
```suggestion
      "version": "1.4.3",
```

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "satellite-visualizer",
-  "version": "1.4.3",
+  "version": "1.5.0",
```
Comment (vishwa2710) [inline package.json:3]:
```suggestion
  "version": "1.4.3",
```

File: src/components/SatelliteVisualizer.tsx [hunk 4]
```diff
@@ -20,11 +20,12 @@ import {
   PolylineDashMaterialProperty,
   IonResource,
   Cartesian2,
+  Matrix3,
 } from 'cesium';
 
 import 'cesium/Build/Cesium/Widgets/widgets.css';
 
-interface Props extends PanelProps<SimpleOptions> {}
+interface Props extends PanelProps<SimpleOptions> { }
```
Comment (vishwa2710) [inline src/components/SatelliteVisualizer.tsx:28]:
```suggestion
interface Props extends PanelProps<SimpleOptions> {}
```

**Comment by lucas-bremond** [comment]
Typo in PR title :)

File: src/components/SatelliteVisualizer.tsx [hunk 5]
```diff
@@ -115,11 +116,35 @@ export const SatelliteVisualizer: React.FC<Props> = ({ options, data, timeRange,
...
 
-        const x_ECEF = Cartesian3.fromDegrees(
-          coalesceToArray(dataFrame.fields[1].values)[i],
-          coalesceToArray(dataFrame.fields[2].values)[i],
-          coalesceToArray(dataFrame.fields[3].values)[i]
-        );
+        const DCM_ECI_ECEF = Transforms.computeFixedToIcrfMatrix(time);
+
+        let x_ECEF: Cartesian3;
+        if (options.coordinatesType === CoordinatesType.cartesianFixed) {
```
Comment (lucas-bremond) [inline src/components/SatelliteVisualizer.tsx:122]: Use `switch` instead?

File: src/types.ts [hunk 6]
```diff
@@ -37,6 +38,12 @@ export enum AssetMode {
   model = 'model',
 }
 
+export enum CoordinatesType {
+  cartesianFixed = 'cartesianFixed',
+  cartesianInertial = 'cartesianInertial',
+  geographic = 'geographic',
```
Comment (lucas-bremond) [inline src/types.ts:44]: Odd enum casing? (check TS conventions)

Comment (lucas-bremond) [inline src/types.ts:44]: Geodetic no?

Comment (vishwa2710) [reply src/types.ts:44]: Pascal case 👍

File: config/grafana.ini [hunk 7]
```diff
@@ -0,0 +1,8 @@
+[plugins]
+allow_loading_unsigned_plugins = lucasbremond-satellitevisualizer-panel
+
+[security]
+allow_embedding = true
+
+[server]
+domain = localhost
```
Comment (lucas-bremond) [inline config/grafana.ini:8]:
```suggestion
domain = localhost

```

**Review by lucas-bremond — APPROVED** [review]

**Pull request merged**